transports/quic/src/connection/substream.rs

```diff
@@ -40,7 +40,10 @@ pub struct SubstreamState {
     /// Waker to wake if the substream becomes closed or stopped.
     pub finished_waker: Option<Waker>,
 
-    /// `true` if the stream finished, i.e. the writing side closed.
+    /// `true` if the writing side of the stream is closing, i.e. `AsyncWrite::poll_close`
+    /// was called.
+    pub is_finishing: bool,
```

Comment:
```suggestion
    pub is_write_closing: bool,
```

Would that not be more intuitive / consistent with `is_write_closed`?

Reply:
An alternative would be to introduce an enum:

``` rust
write_state: WriteState,

// ...

enum WriteState {
  Open,
  Closing,
  Closed,
}
```

I don't have a strong opinion.

Reply:
`finish` is QUIC terminology right?

Reply:
I like `WriteState`. Added in c6256e1.

> `finish` is QUIC terminology right?

Not really QUIC terminology, but used in `quinn`. I stuck with it to avoid confusion on inner events like `StreamEvent::Finished`. But thinking about it again I think using the `AsyncWrite` terminology of `close` is more intuitive.